test/e2e-tests-upgrade.sh

```diff
@@ -16,19 +16,35 @@
 
 # This script calls out to scripts in tektoncd/plumbing to setup a cluster
 # and deploy Tekton Pipelines to it for running upgrading tests. There are
-# two scenarios we need to cover in this script:
+# three scenarios we need to cover in this script:
 
 # Scenario 1: install the previous release, upgrade to the current release, and
 # validate whether the Tekton pipeline works.
 
 # Scenario 2: install the previous release, create the pipelines and tasks, upgrade
 # to the current release, and validate whether the Tekton pipeline works.
 
+# Scenario 3: install the current pipelines, use the integration tests from the previous release(s)
+# to prevent regression. Incompatible changes made in the current release cycle are expected to fail
+# and will be skipped after failing once.
+
 source $(git rev-parse --show-toplevel)/test/e2e-common.sh
 RELEASES='https://github.com/tektoncd/pipeline/releases/latest'
 PREVIOUS_PIPELINE_VERSION=$(curl -L -s -H 'Accept: application/json' $RELEASES |
   sed -e 's/.*"tag_name":"\([^"]*\)".*/\1/')
 
+# BACKWARD_INCOMPATIBLE_TESTS is a manually maintained list of e2e tests that contains known breaking changes
+# made in the current devel cycle. It prevents the upgrade test job from keeping failing after it failed once.
+# TODO: #5782 this struct could be changed to a map once we are testing against more releases
+BACKWARD_INCOMPATIBLE_TESTS=()
+
+function skip_breaking_tests {
+  for TEST in "${BACKWARD_INCOMPATIBLE_TESTS[@]}"; do
+    echo "Remove $TEST since it is a breaking change at devel"
+    rm ./test/"$TEST".go
```

Comment:
It looks like this would remove entire test files, but we might want something more granular. For example, what if we want the ability to remove a single example test?

Reply:
Great point, thanks Lee. I have thought about this with another list of breaking examples tests, does that make sense that we also just remove the breaking ones so that testYamls wouldn't create them?

Also for removing only part of an e2e test, I am not fully sure if that is feasible with our existing plumbing package.

Reply:
I'm not sure I completely understand your question, but I think we'd like to be able to skip individual yaml tests. I'm not sure what you mean by removing part of an e2e test, but it doesn't sound like something we would need.

Reply:
> I'm not sure what you mean by removing part of an e2e test

I wanted to refer to maybe we want to skip [TestHubResolver_Failure](https://github.com/tektoncd/pipeline/blob/897c77709c8bc8fce068868cdd9a4c983b9f34a3/test/resolvers_test.go#LL137C1-L137C29) but not `TestGitResolver_Clone` in https://github.com/tektoncd/pipeline/blob/main/test/resolvers_test.go. It would be a bit complicated if we do have such case.

Reply:
maybe the `go test --run` flag could work here?

Reply:
Thanks, it worked with the skip flag for single tests.